src/index.ts

```diff
@@ -102,6 +100,12 @@ export class HttpClient implements IHttpClient {
     this.got = got.extend(gotOptions);
   }
 
+  // set the method and decide to place the 'json' or 'form' property, or not when no data.
+  #processOptions = (method: Method, data: Record<string, unknown> | undefined, options?: RequestOptions): RequestOptions => {
+    const jsonOrForm = options?.isForm ?? false ? 'form' : data ? 'json' : undefined;
+    return {...(options ?? {}), ...(jsonOrForm ? {[jsonOrForm]: data} : {}), method }
```

Comment:
That works. Maybe consider improving readability a bit by:
```suggestion
    const jsonOrForm = options?.isForm ?? false
        ? 'form'
        : data === undefined
            ? 'json'
            : undefined;
    return {
        ...(options ?? {}),
        ...(jsonOrForm !== undefined
            ? {[jsonOrForm]: data}
            : {}
        ),
        method
    };
```
I also feel we should be more studious in using explicit `null` and `undefined` checks because there are so many bugs that could be lurking here.

Reply:
Yes, it was a bit ninja.. I take your suggestion.